src/main/java/net/neoforged/neoforge/common/IOUtilities.java

```diff
@@ -122,8 +124,13 @@ public static void atomicWrite(Path targetPath, WriteCallback writeCallback) thr
                 TEMP_FILE_SUFFIX);
 
         try {
-            try (OutputStream writeStream = Files.newOutputStream(tempPath, OPEN_OPTIONS)) {
-                writeCallback.write(writeStream);
+            try (var channel = FileChannel.open(tempPath, OPEN_OPTIONS)) {
+                // We need to prevent the callback from closing the channel, since that
+                // would prevent us from flushing it.
+                var stream = CloseShieldOutputStream.wrap((Channels.newOutputStream(channel)));
```

Comment:
We don't need the extra parentheses here; everything else looks fine though :smile: